sdk/include/token.h

```diff
@@ -238,4 +238,18 @@ __always_inline T *token_unseal(SKey key, Sealed<T> sealed)
 {
 	return static_cast<T *>(token_obj_unseal(key, sealed));
 }
+
 #endif // __cplusplus
+
+#if __has_extension(cheri_sealed_pointers)
+#	ifdef __cplusplus
+template<typename T>
+__always_inline T *token_unseal(SKey key, T *__sealed_capability sealed)
+{
+	return static_cast<T *>(token_obj_unseal(key, sealed));
+}
+#	else
+#		define token_unseal(key, sealed) /*NOLINT*/                           \
```

Comment:
Why won't this clash with the `token_unseal` function above, since its `T` is inferrable and so its invocations can be spelled without wakas before parens?

Reply:
Because there is no type that is both a `T *__sealed_capability` and a `Sealed<T>`

Reply:
The macro is visible only in C code.